File: src/leetcode/167twoSum.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int left = 0, right = numbers.size() - 1;
        vector<int> ans;
        while (right > left) {
            if (numbers[left] + numbers[right] > target) {
                right--;
            } else if (numbers[left] + numbers[right] < target) {
                left++;
            } else {
                ans.push_back(left + 1);
                ans.push_back(right + 1);
                break;
            }
        }
        return ans;
    }
};
int main(int argc, char *argv[])
{
    vector<int> numbers = {2, 7, 11, 13};
    int target = 9;
    vector<int> results = Solution().twoSum(numbers, target);
    for(int result:results){
        cout << result << endl;
    }
    return 0;
}

File: scripts/leetcode/167twoSum.py
from typing import List

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers) - 1
        ans = list()
        while left < right:
            if numbers[left] + numbers[right] > target:
                right -= 1
            elif numbers[left] + numbers[right] < target:
                left += 1
            else:
                ans.append(left + 1)
                ans.append(right + 1)
                break
        return ans

numbers = [2,7,11,15]
target = 9
print(f"result: {Solution().twoSum(numbers, target)}")